src/Processors/Formats/Impl/NativeORCBlockInputFormat.cpp

```diff
@@ -97,6 +97,390 @@ std::unique_ptr<orc::InputStream> asORCInputStreamLoadIntoMemory(ReadBuffer & in
     return std::make_unique<ORCInputStreamFromString>(std::move(file_data), file_size);
 }
 
+static const orc::Type * getORCTypeByName(const orc::Type & schema, const String & name)
+{
+    for (uint64_t i = 0; i != schema.getSubtypeCount(); ++i)
+    {
+        if (boost::iequals(schema.getFieldName(i), name))
+            return schema.getSubtype(i);
+    }
+    return nullptr;
+}
+
+static std::optional<orc::PredicateDataType> convertORCTypeToPredicateType(const orc::Type & type)
+{
+    switch (type.getKind())
+    {
+        case orc::BOOLEAN:
+            return orc::PredicateDataType::BOOLEAN;
+        case orc::BYTE:
+        case orc::SHORT:
+        case orc::INT:
+        case orc::LONG:
+            return orc::PredicateDataType::LONG;
+        case orc::FLOAT:
+        case orc::DOUBLE:
+            return orc::PredicateDataType::FLOAT;
+        case orc::VARCHAR:
+        case orc::CHAR:
+        case orc::STRING:
+            return orc::PredicateDataType::STRING;
+        case orc::DATE:
+            return orc::PredicateDataType::DATE;
+        case orc::TIMESTAMP:
+            return orc::PredicateDataType::TIMESTAMP;
+        case orc::DECIMAL:
+            return orc::PredicateDataType::DECIMAL;
+        default:
+            return {};
+    }
+}
+
+static String getColumnNameFromKeyCondition(const KeyCondition & key_condition, size_t indice)
+{
+    const auto & key_columns = key_condition.getKeyColumns();
+    for (const auto & [name, i] : key_columns)
+    {
+        if (i == indice)
+            return name;
+    }
+    throw Exception(ErrorCodes::LOGICAL_ERROR, "Can't get column from KeyCondition with indice {}", indice);
+}
+
+static std::optional<orc::Literal>
+convertFieldToORCLiteral(const orc::Type & orc_type, const Field & field, DataTypePtr type_hint = nullptr)
+{
+    try
+    {
+        /// We always fallback to return null if possible CH type hint not consistent with ORC type
+        switch (orc_type.getKind())
+        {
+            case orc::BOOLEAN: {
+                /// May throw exception
```

Comment:
Looks like it only does the check in debug build: https://github.com/ClickHouse/ClickHouse/blob/68ce6b9b00d5e60de0ed0cbb46a068d2b051f921/src/Core/Field.h#L857-L881

Replace the two `get()`s with `tryGet()`s and remove the try-catch?

Reply:
`tryGet()` has more stricter type checking than `get()`, they are quite different when I want to read `Int64` from a field with type `UInt64`.  `get()` allows this behavior but `tryGet()` throws exception.

For example: filter condition like `a > 40`.  Literal `40` has type `UInt64`, but `a` has type `Int64`. We want to convert field `40` with `UInt64` type to orc::Literal with LONG type.